morph_tool/resampling.py

```diff
@@ -170,13 +163,18 @@ def _dispatch_section_function(cell_family):
     of the morphology
     """
     return {
-        morphio.CellFamily.FAMILY_NEURON: _resample_section_neuron,
-        morphio.CellFamily.FAMILY_GLIA: _resample_section_astrocyte
+        morphio.CellFamily.FAMILY_NEURON: _resample_neuron_section,
+        morphio.CellFamily.FAMILY_GLIA: _resample_astrocyte_section
     }[cell_family]
 
 
 def resample_linear_density(obj, linear_density):
-    """Resample the number of points in morphology
+    """Returns a new morphology with new points and point properties,
+    the number of which depends on the linear_density. The new
+    values are linearly interpolated from the old ones.
+
+    This function is useful for use cases where morphologies have too
+    many points and a resampling with fewer points/properties is required.
```

Comment:
I would have bet that the goal was to increase the number of points! Good to know.

Reply:
Topological astrocyte synthesis has a very small segment length to capture the branching of very small sections. But the resulting morphology has unecessarily too many segments in the big sections. Resampling after synthesizing will resolve this issue. Unfortunately, there is no easy way to make the segment length adaptive in synthesis for the time being.